File: mesop/examples/web_component/copy_to_clipboard/copy_to_clipboard_component.py
import mesop as me


@me.web_component(path="./copy_to_clipboard_component.js")
def copy_to_clipboard_component(
  *,
  text: str = "",
  key: str | None = None,
):
  return me.insert_web_component(
    name="copy-to-clipboard-component",
    key=key,
    properties={
      "text": text,
    },
  )

File: mesop/examples/web_component/copy_to_clipboard/copy_to_clipboard_component.js
import {
  LitElement,
  html,
} from 'https://cdn.jsdelivr.net/gh/lit/dist@3/core/lit-core.min.js';

class CopyToClipboard extends LitElement {
  static properties = {
    text: {type: String},
  };

  constructor() {
    super();
    this.text = '';
  }

  render() {
    return html`
      <div @click="${this._onClick}">
        <slot></slot>
      </div>
    `;
  }

  _onClick() {
    navigator.clipboard
      .writeText(this.text)
      .catch((err) => console.error('Failed to copy text: ', err));
  }
}

customElements.define('copy-to-clipboard-component', CopyToClipboard);
